src/shouldFallbackToLegacyNativeModule.js

```diff
@@ -0,0 +1,20 @@
+const {NativeModules} = require('react-native');
+
+export default function shouldFallbackToLegacyNativeModule() {
+  const expoExecutionEnvironment = NativeModules.NativeUnimoduleProxy?.modulesConstants?.ExponentConstants?.executionEnvironment;
+
+  /**
+   * Expo managed apps don't include the @react-native-async-storage/async-storage
+   * native modules yet, but the API interface is the same, so we can use the version
+   * exported from React Native still.
+   *
+   * If in future releases (eg: @react-native-async-storage/async-storage >= 2.0.0) this
+   * will likely not be valid anymore, and the package will need to be included in the Expo SDK
+   * to continue to work.
+   */
+  if (expoExecutionEnvironment && ['storeClient', 'standalone'].includes(expoExecutionEnvironment)) {
```

Comment:
@brentvatne 
Are those execution envs available in all expo versions? Or is there a min version support for it?

Reply:
only in sdk 40+ -- but a version of async storage that includes this code will only ever run in sdk 41+ apps, because users install a compatible version of native modules for given sdks using `expo install`